zerver/lib/actions.py

```diff
@@ -5127,6 +5129,23 @@ def do_update_user_custom_profile_data(user_profile: UserProfile,
                 "rendered_value": field_value.rendered_value,
                 "type": field_value.field.field_type})
 
+def do_check_remove_custom_profile_field_value(user_profile: UserProfile,
+                                               field_id: Union[int, str, List[int]]
+                                               ) -> Union[bool, HttpResponse]:
+    try:
+        field = CustomProfileField.objects.get(realm=user_profile.realm, id=field_id)
+        field_value = CustomProfileFieldValue.objects.get(field=field, user_profile=user_profile)
+        field_value.delete()
+        notify_user_update_custom_profile_data(user_profile, {'id': field_id,
+                                                              'value': None,
+                                                              'rendered_value': None,
+                                                              'type': field.field_type})
+    except CustomProfileField.DoesNotExist:
+        return json_error(_('Field id {id} not found.').format(id=field_id))
```

Comment:
This should `raise JsonableError()` instead -- that will do the same thing as this `json_error` code path, without the caller needing to check the return value from this function.